gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1308 @@
+
+        # keeping only VALID cores
+        valid_core_mask = np.vectorize(self._is_valid_core)(results)
+        valid_topics = self.ttda[valid_core_mask]
+        topic_labels = np.array([topic["label"] for topic in results])[valid_core_mask]
+        unique_labels = np.unique(topic_labels)
+
+        num_stable_topics = len(unique_labels)
+        stable_topics = np.empty((num_stable_topics, len(self.id2word)))
+
+        # for each cluster
+        for l, label in enumerate(unique_labels):
+            # mean of all the topics that are of that cluster
+            topics_of_cluster = np.array([topic for t, topic in enumerate(valid_topics) if topic_labels[t] == label])
+            stable_topics[l] = topics_of_cluster.mean(axis=0)
+
+        self.sorted_clusters = sorted_clusters
+        self.stable_topics = stable_topics
+
+        logger.info("found %s stable topics", len(stable_topics))
+
+    def recluster(self, eps=0.1, min_samples=None, min_cores=None):
+        """Reapply CBDBSCAN clustering and stable topic generation.
+
+        Stable topics can be retrieved using :meth:`~gensim.models.ensemblelda.EnsembleLda.get_topics`.
+
+        Parameters
+        ----------
+        eps : float
+            epsilon for the CBDBSCAN algorithm, having the same meaning as in classic DBSCAN clustering.
+            default: ``0.1``
+        min_samples : int
+            The minimum number of samples in the neighborhood of a topic to be considered a core in CBDBSCAN.
+            default: ``int(self.num_models / 2)``
+        min_cores : int
+            how many cores a cluster has to have, to be treated as stable topic. That means, how many topics
+            that look similar have to be present, so that the average topic in those is used as stable topic.
+            default: ``min(3, max(1, int(self.num_models /4 +1)))``
+
+        """
+        # if new models were added to the ensemble, the distance matrix needs to be generated again
+        if self.asymmetric_distance_matrix_outdated:
+            logger.info("asymmetric distance matrix is outdated due to add_model")
+            self._generate_asymmetric_distance_matrix()
+
+        # Run CBDBSCAN to get topic clusters:
+        self._generate_topic_clusters(eps, min_samples)
+
+        # Interpret the results of CBDBSCAN to identify stable topics:
+        self._generate_stable_topics(min_cores)
+
+        # Create gensim LdaModel representation of topic model with stable topics (can be used for inference):
+        self.generate_gensim_representation()
+
+    # GENSIM API
+    # to make using the ensemble in place of a gensim model as easy as possible
+
+    def get_topics(self):
+        """Return only the stable topics from the ensemble.
+
+        Returns
+        -------
+        2D Numpy.numpy.ndarray of floats
+            List of stable topic term distributions
+
+        """
+        return self.stable_topics
+
+    def _has_gensim_representation(self):
```

Comment:
The name is a bit misleading: it feels like a yes/no question and hints that it may return a boolean. Something like this:

```suggestion
    def _ensure_gensim_representation(self):
```

It's almost like an assert, with the exception that it's raising a ValueError instead of an AssertionError.